src/main/java/net/imglib2/algorithm/stats/InformationMetrics.java

```diff
@@ -0,0 +1,218 @@
+package net.imglib2.algorithm.stats;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.imglib2.Cursor;
+import net.imglib2.IterableInterval;
+import net.imglib2.histogram.BinMapper1d;
+import net.imglib2.histogram.Histogram1d;
+import net.imglib2.histogram.HistogramNd;
+import net.imglib2.histogram.Real1dBinMapper;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.type.numeric.integer.LongType;
+import net.imglib2.view.IntervalView;
+import net.imglib2.view.Views;
+
+/**
+ * This class provides method for computing information metrics (entropy, mutual information)
+ * for imglib2.
+ * 
+ * @author John Bogovic
+ */
+public class InformationMetrics
+{
+
+	/**
+	 * Returns the normalized mutual information of the inputs
+	 * @param rai the RandomAccessibleInterval
+	 * @param ra the RandomAccessible
+	 * @return the normalized mutual information
+	 */
+	public static <T extends RealType< T >> double normalizedMutualInformation(
+			final IterableInterval< T > dataA,
+			final IterableInterval< T > dataB,
+			double histmin, double histmax, int numBins )
+	{
+		HistogramNd< T > jointHist = jointHistogram( dataA, dataB, histmin, histmax, numBins );
+		return normalizedMutualInformation( jointHist );
+	}
+
+	/**
+	 * Returns the normalized mutual information of the inputs
+	 * @param jointHist the joint histogram
+	 * @return the normalized mutual information
+	 */
+	public static <T extends RealType< T >> double normalizedMutualInformation(
+		final HistogramNd< T > jointHist )
+	{
+		double HA = marginalEntropy( jointHist, 0 );
+		double HB = marginalEntropy( jointHist, 1 );
+		double HAB = entropy( jointHist );
+		return 2 * ( HA + HB - HAB ) / ( HA + HB );
+	}
+	
+	/**
+	 * Returns the normalized mutual information of the inputs
+	 * @param rai the RandomAccessibleInterval
+	 * @param ra the RandomAccessible
+	 * @return the normalized mutual information
+	 */
+	public static <T extends RealType< T >> double mutualInformation(
+			IterableInterval< T > dataA,
+			IterableInterval< T > dataB,
+			double histmin, double histmax, int numBins )
+	{
+		HistogramNd< T > jointHist = jointHistogram( dataA, dataB, histmin, histmax, numBins );
+
+		double HA = marginalEntropy( jointHist, 0 );
+		double HB = marginalEntropy( jointHist, 1 );
+		double HAB = entropy( jointHist );
+
+		return HA + HB - HAB;
```

Comment:
This should call `mutualInformation( jointHist )` instead of duplicating the code.